src/psyclone/psyad/tl2ad.py

```diff
@@ -268,6 +269,54 @@ def generate_adjoint(tl_psyir, active_variables):
     return ad_psyir
 
 
+def _add_precision_symbol(symbol, table):
+    '''
+    Adds (a copy of) the supplied precision symbol to the supplied symbol
+    table unless there is already an entry with the same name. Also takes
+    care of any ContainerSymbol from which the symbol is imported.
+
+    :param symbol: the precision symbol to copy into the table.
+    :type symbol: :py:class:`psyclone.psyir.symbols.DataSymbol`
+    :param table: the symbol table to which to add the precision symbol.
+    :type table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+
+    :raises TypeError: if the supplied symbol is not of the correct type.
+    :raises NotImplementedError: if the supplied symbol is not local or \
+                                 explicitly imported.
+
+    '''
+    # A precision symbol must be of integer, deferred or unknown type.
+    if not (isinstance(symbol.datatype, (DeferredType, UnknownType)) or
+            isinstance(symbol.datatype, ScalarType) and
+            symbol.datatype.intrinsic == ScalarType.Intrinsic.INTEGER):
+        raise TypeError(
```

Comment:
Is it possible to update the tests so that at least one tests the full string including the full stop. They are on lines 566 and 570 of tl2ad_test.py. My guess is that there was a difference between Python2 and 3 which caused us to only check a subset. If there is a difference between Python3 versions then please ignore the request.

Reply:
Done, let's see what the CI says...